src/history/hs_rec.c

```diff
@@ -526,24 +534,6 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_MULTI *mult
         prev_upd = upd = NULL;
 
         WT_ASSERT(session, updates.size > 0);
-        __wt_update_vector_peek(&updates, &oldest_upd);
```

Comment:
Why we remove this code? I think it is still needed?

Reply:
Suppose a case that we have U20 -> U30 -> T0 onpage, do we still need this code?

Reply:
In the above scenario, the data store is U3(0 -0) inserted and the history store is inserted with U2(0-0), so there is no need for clearing the history store contents when we detect the last update is a tombstone.

Earlier the U2 was inserting with (20-30) time window, now we fix it during the insert itself, so there is no need to remove.

Reply:
Why the data store is U3(0-0), I don't understand stand. We can have a chat tomorrow.

Reply:
Scenario - T(0) -> U3 (30) -> U2(20)
We do not write anything for that key if the tombstone is globally visible. In case the tombstone is not globally visible we will write U3 (0 -0) time window to the data store and without this fix, we write U2(20-30) to the history store.
Due to this reason, we have to remove the history store updates when the tombstone is globally visible.


With this fix, we write U2 to the history store with a (0-0) time window, so there is no need to remove the history store contents when the tombstone is globally visible.

Reply:
I'm talking the scenario that we reconcile after put a T(0), so we have U30 -> U20 -> T0 on the update chain, onpage may be U10

Reply:
Thanks. I understood the scenario. Added back the change. Will add a test case to cover this scenario.

Reply:
Added test to test the above scenario, so we will get a failure whenever we remove that part of the code along with the test to test this scenario.